File: Medium/452. Minimum Number of Arrows to Burst Balloons/MinimumNumberOfArrowsToBurstBalloons.java
/*
LeetCode Medium: 452. Minimum Number of Arrows to Burst Balloons
https://leetcode.com/problems/minimum-number-of-arrows-to-burst-balloons/

Check MinimumNumberOfArrowsToBurstBalloons.cpp for brief solution explanation.
*/
class Solution {
    public int findMinArrowShots(int[][] points) {
        int N = points.length;
        if(N==0){
            return 0;
        }
        Arrays.sort(points, (a, b) -> {
            if(a[1]<b[1]){
                return -1;
            }else if(a[1]==b[1]){
                return 0;
            }else{
                return 1;
            }
        });
        int[]cur = points[0];
        int ans = 1;
        for(int i = 1;i<N;i++){
 
            if(cur[1]<points[i][0]){
                ans++;
                cur = points[i];
            }
        }
        return ans;
    }
}

File: Medium/452. Minimum Number of Arrows to Burst Balloons/MinimumNumberOfArrowsToBurstBalloons.py
'''
LeetCode Medium: 452. Minimum Number of Arrows to Burst Balloons
https://leetcode.com/problems/minimum-number-of-arrows-to-burst-balloons/

Check MinimumNumberOfArrowsToBurstBalloons.cpp for brief solution explanation.
'''
class Solution:
    def findMinArrowShots(self, points):
        points.sort(key = lambda x: x[1])
        N = len(points)
        if N==0:
            return 0
        ans = 0
        cur = points[0]
        for i in range(1, n):
            if cur[1]<points[i][0]:
                ans+=1
                cur = points[i]
        return ans